site/src/pages/WorkspacePage/WorkspaceScheduleControls.tsx

```diff
@@ -0,0 +1,387 @@
+    ...decreaseDeadline(workspace),
+    onSuccess: onDeadlineChangeSuccess,
+    onError: onDeadlineChangeFails,
+  });
+  const increaseMutation = useMutation({
+    ...increaseDeadline(workspace),
+    onSuccess: onDeadlineChangeSuccess,
+    onError: onDeadlineChangeFails,
+  });
+
+  return (
+    <div css={styles.scheduleValue}>
+      {isWorkspaceOn(workspace) ? (
+        <AutoStopDisplay workspace={workspace} />
+      ) : (
+        <ScheduleSettingsLink>
+          {autostartDisplay(workspace.autostart_schedule)}
+        </ScheduleSettingsLink>
+      )}
+
+      {canUpdateSchedule && canEditDeadline(workspace) && (
+        <span css={styles.scheduleControls}>
+          <Popover>
+            <PopoverTrigger>
+              <IconButton
+                disabled={!deadlineMinusEnabled}
+                size="small"
+                title="Subtract hours from deadline"
+                css={styles.scheduleButton}
+              >
+                <RemoveIcon />
+              </IconButton>
+            </PopoverTrigger>
+            <PopoverContent
+              id="schedule-sub"
+              classes={{ paper: classNames.paper }}
+              horizontal="right"
+            >
+              <DecreaseTimeContent
+                maxDeadlineDecrease={maxDeadlineDecrease}
+                onDeadlineMinus={decreaseMutation.mutate}
+              />
+            </PopoverContent>
+          </Popover>
+          <Popover>
+            <PopoverTrigger>
+              <IconButton
+                disabled={!deadlinePlusEnabled}
+                size="small"
+                title="Add hours to deadline"
+                css={styles.scheduleButton}
+              >
+                <AddIcon />
+              </IconButton>
+            </PopoverTrigger>
+            <PopoverContent
+              id="schedule-add"
+              classes={{ paper: classNames.paper }}
+              horizontal="right"
+            >
+              <AddTimeContent
+                maxDeadlineIncrease={maxDeadlineIncrease}
+                onDeadlinePlus={increaseMutation.mutate}
+              />
+            </PopoverContent>
+          </Popover>
+        </span>
+      )}
+    </div>
+  );
+};
+
+interface AddTimeContentProps {
+  maxDeadlineIncrease: number;
+  onDeadlinePlus: (value: number) => void;
+}
+
+const AddTimeContent: FC<AddTimeContentProps> = ({
+  maxDeadlineIncrease,
+  onDeadlinePlus,
+}) => {
+  const popover = usePopover();
+
+  return (
+    <>
+      <span css={styles.timePopoverTitle}>Add hours to deadline</span>
+      <span css={styles.timePopoverDescription}>
+        Delay the shutdown of this workspace for a few more hours. This is only
+        applied once.
+      </span>
+      <form
+        css={styles.timePopoverForm}
+        onSubmit={(e) => {
+          e.preventDefault();
+          const formData = new FormData(e.currentTarget);
+          const hours = Number(formData.get("hours"));
```

Comment:
I don't _think_ this needs to be fixed, but I just want to make sure I'm understanding what would happen with an accident like this:
1. Someone moves things elements around, and the `name` field accidentally gets removed
2. `formData.get("hours")` evaluates to `null`, and `Number` converts it into `0`

It looks like it would make a PUT request with `0`, which I'm guessing wouldn't break anything, but just want to make sure

Reply:
You are exactly right since increase or decrease something in 0 would result in the same value, but I love your analysis